engine/src/main/java/org/terasology/logic/players/CameraClientSystem.java

```diff
@@ -29,12 +29,30 @@
 import org.terasology.logic.location.Location;
 import org.terasology.logic.location.LocationComponent;
 import org.terasology.logic.permission.PermissionManager;
+import org.terasology.logic.players.event.OnPlayerSpawnedEvent;
 import org.terasology.logic.players.event.ResetCameraEvent;
 import org.terasology.math.geom.Quat4f;
 import org.terasology.math.geom.Vector3f;
 import org.terasology.network.ClientComponent;
 import org.terasology.registry.In;
 
+/**
+ * This is a system that creates and maintains a client side entity for the camera.
+ * <p>
+ * Entity Creation Flow:
+ * - ClientComponent for the local player added or changed
+ * - If Camera does not exist, create "engine:camera" entity
+ * - Save the ClientComponent with the new camera attached
+ * <p>
+ * Reset Camera Event Flow:
+ * - ResetCameraEvent
+ * - Camera entity on the ClientComponent is destroyed and set to EntityRef.NULL
+ * - ClientComponent is saved
+ * <p>
+ * Auto Mount Camera Flow:
+ * - ClientComponent or AutoMountCameraComponent for the local player added or changed
+ * - The local player's camera entity (created from above steps) is location linked to the Gaze Entity
+ */
 @RegisterSystem(RegisterMode.CLIENT)
```

Comment:
Correct. These descriptions are vague.  My intent was to generally outline what the code does so that there is a general idea of what to look for when perusing the code.
Possibly I have taken the summary too far in an attempt to improve skimmability by reducing word count and putting the target of the step at the front... Dunno.